cli_reference/extend_cli.adoc

```diff
@@ -0,0 +1,218 @@
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+See the link:https://access.redhat.com/support/offerings/techpreview/[Red Hat
+Technology Preview features support scope] for more information.
+endif::[]
+====
+endif::[]
+
+[[cli-plugins-prerequisites]]
+== Prerequisites
+
+You must have:
+
+- A xref:../cli_reference/get_started_cli.adoc#cli-reference-get-started-cli[working `oc` binary installed].
+- An `oc` version of 3.7 or later (recommended).
+
+[[cli-installing-plugins]]
+== Installing Plug-ins
+
+Copy the plug-in's *_plugin.yaml_* descriptor, binaries, scripts, and assets
+files to one of the locations in the file system where `oc` searches for
+plug-ins.
+
+Currently, {product-title} does not provide a package manager for plug-ins.
+Therefore, it is your responsibility to place the plug-in files in the correct
+location. It is recommended that each plug-in is located on its own directory.
+
+To install a plug-in that is distributed as a compressed file, extract it to one
+of the locations specified in xref:cli-plugin-loader[The Plug-in Loader] section.
+
+[[cli-plugin-loader]]
+=== The Plug-in Loader
+
+The plug-in loader is responsible for
+xref:cli-plugin-loader-search-order[searching plug-in files], and checking if
+the plug-in provides the minimum amount of information required for it to run.
+Files placed in the correct location that do not provide the minimum amount of
+information (for example, an incomplete *_plugin.yaml_* descriptor) are ignored.
+
+[[cli-plugin-loader-search-order]]
+==== Search Order
+The plug-in loader uses the following search order:
+
+. `${KUBECTL_PLUGINS_PATH}` If specified, the search stops here.
+. `${XDG_DATA_DIRS}/kubectl/plugins`
+. `~/.kube/plugins`
+
+If the `KUBECTL_PLUGINS_PATH` environment variable is present, the loader uses
+it as the only location to look for plug-ins. The `KUBECTL_PLUGINS_PATH`
+environment variable is a list of directories. In Linux and Mac, the list is
+colon-delimited. In Windows, the list is semicolon-delimited.
+
+If `KUBECTL_PLUGINS_PATH` is not present, the loader searches these additional locations:
+
+. One or more directories specified according to the
+link:https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html[XDG
+System Directory Structure] specification.
++
+Specifically, the loader locates the directories specified by the
+`XDG_DATA_DIRS` environment variable, and then searches the *_kubectl/plugins_*
+directory inside of those. If `XDG_DATA_DIRS` is not specified, it defaults to
+*_/usr/local/share:/usr/share_*.
+
+. The `plugins` directory under the user's *_kubeconfig_* directory. In most
+cases, this is *_~/.kube/plugins_*:
++
+----
+# Loads plugins from both /path/to/dir1 and /path/to/dir2
+$ KUBECTL_PLUGINS_PATH=/path/to/dir1:/path/to/dir2 kubectl plugin -h
+----
```

Comment:
I think there's some redundancy in the above section, in that there's the order, but then the individual sections are talked about, with only the second two using numbered points. Can I suggest something like:

> The plug-in loader uses the following search order:
> 
> . `${KUBECTL_PLUGINS_PATH}`
> 
> <info>
> 
> . `${XDG_DATA_DIRS}/kubectl/plugins`
> 
> <info>
> 
> . `~/.kube/plugins`
> 
> <info>

I hope this makes sense. And also, just a suggestions :)